test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/CompareAttributeAdapterTest.cs

```diff
@@ -25,19 +26,30 @@ public void ClientRulesWithCompareAttribute_ErrorMessageUsesDisplayName_WithoutL
             var attribute = new CompareAttribute("OtherProperty");
             var adapter = new CompareAttributeAdapter(attribute, stringLocalizer: null);
 
+            // Mono issue - https://github.com/aspnet/External/issues/19
+            var expectedMessage = PlatformNormalizer.NormalizeContent(
+                    "'MyPropertyDisplayName' and 'OtherPropertyDisplayName' do not match.");
+
             var actionContext = new ActionContext();
-            var context = new ClientModelValidationContext(actionContext, metadata, metadataProvider);
+            var context = new ClientModelValidationContext(
+                actionContext,
+                metadata,
+                metadataProvider,
+                new AttributeDictionary());
```

Comment:
Add tests covering non-empty `AttributeDictionary`s e.g. one that already contains `data-val = false` and `attribute-name = "attribute-value"`. Fine to spread this out a bit between `BlahAdapterTest` classes and to _not_ cover every single attribute the adapters add.